### PR TITLE
[GPU] adding weight format conversion rule for b_fs_yx_fsv16

### PR DESCRIPTION
**Pull request by e-ddykim**
### Details:
 - updated `to_weights_format()` to convert `b_fs_yx_fsv16` to `o_is_yx_isv16`

### Tickets:
 - 89449

### CODE REVIEW

File: src/plugins/intel_gpu/src/runtime/layout.cpp [hunk 1]
```diff
@@ -214,6 +214,8 @@ static format to_weights_format(format f, bool is_grouped) {
                 throw std::runtime_error("Invalid conversion of data format to weights format. bfwzyx can't be non-grouped as 4D spatials are not supported");
             return format::goizyx;
         }
+        case format::b_fs_yx_fsv16:
+            return format::o_is_yx_isv16;
```
Comment (yeonbok) [inline src/plugins/intel_gpu/src/runtime/layout.cpp:218]: Just want to check: Is the weight tensor of the original target problem not from the constant weight but from the other normal operation? 
Then, I think we should add reorder between the weight tensor and the convolution.
Could you please check the detailed situation?

Comment (yeonbok) [reply src/plugins/intel_gpu/src/runtime/layout.cpp:218]: Please refer to this commit: b492b591363fe5493b02656a86ab186c910b00ab

**Comment by akladiev** [comment]
The valid link to the validation results: https://openvino-ci.toolbox.iotg.sclab.intel.com/job/private-ci/job/github_trigger/job/openvino/job/PR-12599/1/

File: src/plugins/intel_gpu/src/runtime/layout.cpp (hunk 1 above)
Comment (yeonbok) [reply src/plugins/intel_gpu/src/runtime/layout.cpp:218]: Checked offline: the preferred weight format is correctly added by post optimize weight pass

**Pull request closed without merging**